apps/src/p5lab/spritelab/SpritelabInput.jsx

```diff
@@ -64,30 +74,84 @@ class SpritelabInput extends React.Component {
     collapsed: false
   };
 
-  userInputSubmit() {
+  userInputSubmit(value) {
     const variableName =
       this.props.inputList[0] && this.props.inputList[0].variableName;
     if (!variableName) {
       return;
     }
     this.props.onPromptAnswer();
-    coreLibrary.onPromptAnswer(variableName, this.state.userInput);
+    coreLibrary.onPromptAnswer(variableName, value);
+  }
+
+  onTextSubmit() {
+    this.userInputSubmit(this.state.userInput);
     this.setState({userInput: ''});
   }
 
+  onMultipleChoiceSubmit(e) {
+    this.userInputSubmit(e.target.value);
+  }
+
   toggleCollapsed = () =>
     this.setState({
       collapsed: !this.state.collapsed
     });
 
   render() {
+    const inputInfo = this.props.inputList[0];
+    if (!inputInfo) {
+      return null;
+    }
     const icon = this.state.collapsed ? 'angle-right' : 'angle-down';
     const numPrompts = this.props.inputList.length;
-    const promptText =
-      this.props.inputList[0] && this.props.inputList[0].promptText;
-    if (!promptText) {
-      return null;
+    const promptText = inputInfo.promptText;
+
+    let inputRow;
+    switch (inputInfo.promptType) {
+      case PromptType.TEXT:
+        inputRow = (
+          <div style={styles.inputRow}>
+            <input
+              style={styles.inputArea}
+              type="text"
+              onChange={event => this.setState({userInput: event.target.value})}
+              value={this.state.userInput || ''}
+            />
+            <button
+              style={styles.submitButton}
+              type="button"
+              onClick={this.onTextSubmit.bind(this)}
+            >
+              <i className="fa fa-check" />
+            </button>
+          </div>
+        );
+        break;
+      case PromptType.MULTIPLE_CHOICE:
+        inputRow = (
+          <div style={styles.inputRow}>
+            {inputInfo.choices.map(
+              choice =>
+                choice && (
+                  <button
+                    key={choice}
+                    style={styles.choiceButton}
+                    type="button"
+                    value={choice}
+                    onClick={this.onMultipleChoiceSubmit.bind(this)}
```

Comment:
super nit:
```suggestion
                    onClick={this.onMultipleChoiceSubmit}
```

Reply:
That's not the same functionality though. We need to bind `this` for the handler so that the scope is right when the handler is called. (same comment below)

Reply:
if your handlers use fat-arrow syntax (`functionName = () => {}` instead of `functionName() {}`) they will auto-bind